build/config/BUILDCONFIG.gn

```diff
@@ -632,6 +632,30 @@ if (custom_toolchain != "") {
   }
 }
 
+# Sets default dependencies for executable and shared_library targets.
+#
+# Variables
+#   no_default_deps: If true, no standard dependencies will be added.
+foreach(_target_type,
```

Comment:
This makes adding implicit dependencies to libcxx obsolete (i.e. this line can now be removed: https://github.com/flutter/engine/pull/6886/files#diff-633f71f2d7c61833a0d6668534227b21R63).

Reply:
This needs to be on Android only for now.

Reply:
I had the `is_android` check down below, but to make it more clear I pulled it up to the top level.

Reply:
Derp. Didn't notice it down below. Yeah, moving it to the top level with a comment saying why this is android only would be great.